src/bp/core/api.ts

```diff
@@ -72,6 +73,25 @@ const event = (eventEngine: EventEngine, eventRepo: EventRepository): typeof sdk
   }
 }
 
+const messaging = (messagingApi: MessagingAPI): typeof sdk.messaging => {
+  return {
+    getAllConversations: messagingApi.getAllConversations.bind(messagingApi),
+    getRecentConversations: messagingApi.getRecentConversations.bind(messagingApi),
+    deleteAllConversations: messagingApi.deleteAllConversations.bind(messagingApi),
+    createConversation: messagingApi.createConversation.bind(messagingApi),
+    getOrCreateRecentConversation: messagingApi.getOrCreateRecentConversation.bind(messagingApi),
+    getConversationById: messagingApi.getConversationById.bind(messagingApi),
+    deleteConversation: messagingApi.deleteConversation.bind(messagingApi),
+    getRecentMessages: messagingApi.getRecentMessages.bind(messagingApi),
+    deleteAllMessages: messagingApi.deleteAllMessages.bind(messagingApi),
+    createMessage: messagingApi.createMessage.bind(messagingApi),
+    getMessageById: messagingApi.getMessageById.bind(messagingApi),
+    deleteMessage: messagingApi.deleteMessage.bind(messagingApi),
+    sendIncoming: messagingApi.sendIncoming.bind(messagingApi),
+    sendOutgoing: messagingApi.sendOutgoing.bind(messagingApi)
+  }
```

Comment:
Maybe we could have two different namespaces instead

So : 
```ts
bp.message.send()
bp.conversation.getOrCreate()
```

This would avoid all the wordiness of having to type `bp.messaging.conversations.getOrCreate`

Reply:
Here's my proposed renaming :
```ts
const message = (messagingApi: MessagingAPI): typeof sdk.message => {
  return {
    many: {
      recent: messagingApi.getRecentMessages.bind(messagingApi),
      delete: messagingApi.deleteAllMessages.bind(messagingApi)
    },
    create: messagingApi.createMessage.bind(messagingApi),
    byId: messagingApi.getMessageById.bind(messagingApi),
    delete: messagingApi.deleteMessage.bind(messagingApi),
    send: messagingApi.sendOutgoing.bind(messagingApi),
    receive: messagingApi.sendIncoming.bind(messagingApi)
  }
}

const conversation = (messagingApi: MessagingAPI): typeof sdk.conversation => {
  return {
    many: {
      all: messagingApi.getAllConversations.bind(messagingApi),
      recent: messagingApi.getRecentConversations.bind(messagingApi),
      delete: messagingApi.deleteConversation.bind(messagingApi)
    },
    create: messagingApi.createConversation.bind(messagingApi),
    for: messagingApi.getOrCreateRecentConversation.bind(messagingApi),
    byId: messagingApi.getConversationById.bind(messagingApi),
    delete: messagingApi.deleteConversation.bind(messagingApi)
  }
}
```

So you'd get something like : 
```ts
const conversation = bp.conversation.for({ userId, botId })
const message = bp.message.send(conversation.id, payload)
```
So the most notable changes would be that `getOrCreate` becomes `for`, `sendOutgoing` becomes `send` and sendIncoming becomes `receive`.

Then any function that has anything to do with multiple items is under `.many` so :
```ts
const recentConversations = bp.conversation.many.recent({ userId, botId })
const lastMessages = bp.message.many.recent(conversation.id, 100)
bp.message.many.delete(conversation.id)
```
And we also avoid using `get` and `set` in the function names